lerobot/common/datasets/transforms.py

```diff
@@ -0,0 +1,118 @@
+        n_subset (int or None): number of transformations to apply. If ``None``,
+            all transforms are applied.
+        random_order (bool): apply transformations in a random order
+    """
+
+    def __init__(
+        self,
+        transforms: Sequence[Callable],
+        p: list[float] | None = None,
+        n_subset: int | None = None,
+        random_order: bool = False,
+    ) -> None:
+        super().__init__()
+        if not isinstance(transforms, Sequence):
+            raise TypeError("Argument transforms should be a sequence of callables")
+        if p is None:
+            p = [1] * len(transforms)
+        elif len(p) != len(transforms):
+            raise ValueError(
+                f"Length of p doesn't match the number of transforms: {len(p)} != {len(transforms)}"
+            )
+
+        if n_subset is None:
+            n_subset = len(transforms)
+        elif not isinstance(n_subset, int):
+            raise TypeError("n_subset should be an int or None")
+        elif not (0 <= n_subset <= len(transforms)):
+            raise ValueError(f"n_subset should be in the interval [0, {len(transforms)}]")
+
+        self.transforms = transforms
+        self.n_subset = n_subset
+        self.random_order = random_order
+
+    def forward(self, *inputs: Any) -> Any:
+        needs_unpacking = len(inputs) > 1
+
+        indices = torch.arange(len(self.transforms))
+        selected_indices = torch.randperm(len(indices))[: self.n_subset]
+        if not self.random_order:
+            selected_indices = selected_indices.sort().values
+
+        selected_transforms = [self.transforms[i] for i in selected_indices]
+
+        for transform in selected_transforms:
+            outputs = transform(*inputs)
+            inputs = outputs if needs_unpacking else (outputs,)
+
+        return outputs
+
+    def extra_repr(self) -> str:
+        format_string = [f"N={self.N}"]
+        for t in self.transforms:
+            format_string.append(f"    {t}")
+        return "\n".join(format_string)
+
+
+class RangeRandomSharpness(Transform):
+    """Similar to RandomAdjustSharpness but with p=1 and a sharpness_factor sampled randomly
+    each time in [range_min, range_max].
+
+    If the input is a :class:`torch.Tensor`,
+    it is expected to have [..., 1 or 3, H, W] shape, where ... means an arbitrary number of leading dimensions.
+    """
+
+    def __init__(self, range_min: float, range_max) -> None:
+        super().__init__()
+        self.range_min, self.range_max = self._check_input(range_min, range_max)
+
+    def _check_input(self, range_min, range_max):
+        if range_min < 0:
+            raise ValueError("range_min must be non negative.")
+        if range_min > range_max:
+            raise ValueError("range_max must greater or equal to range_min")
+        return range_min, range_max
+
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
```

Comment:
Only `forward` is required since `Transform` is just a wrapper around `nn.Module`. `RandomSubsetApply` is not really a transform and is more akin to `v2.Compose` so I've implemented it in a similar way (with `forward`) while `RangeRandomSharpness` (now `SharpnessJitter`) is actually a transform and can use the `forward` from its parent `Tranform` which in turn calls `_transformed` (at least that's how I understand it)